build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -211,7 +212,9 @@ class _SingleBuild {
     var watch = Stopwatch()..start();
     var result = await _safeBuild(updates);
     var optionalOutputTracker = OptionalOutputTracker(
-        _assetGraph, _outputLocations.keys.toList(), _buildPhases);
+        _assetGraph,
+        _outputLocations.keys.where((k) => k.isNotEmpty).toList(),
```

Comment:
This feels wrong to me. I'd expect something like

`_outputLocations.keys.contains('') ? [] : _outputLocations.keys.toList(),`